frontend/src/Student/components/OtherMediaSubmissionForm.js

```diff
@@ -0,0 +1,287 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { Link } from 'react-router-dom'
+import { Form, FormGroup, FormFeedback, Label, Button, Row, Col } from 'reactstrap'
+import styled from 'styled-components'
+import Dropzone from 'react-dropzone'
+import { Formik, Field } from 'formik'
+import yup from 'yup'
+
+const Header = styled.h1`
+  margin-bottom: 25px;
+`
+
+const PreviewImage = styled.img`
+  height: 100%;
+`
+
+const ButtonContainer = styled.div`
+  margin-top: 50px;
+`
+
+class OtherSubmissionForm extends Component {
+  static propTypes = {
+    user: PropTypes.shape({
+      username: PropTypes.string
+    }).isRequired,
+    handleImageUpload: PropTypes.func.isRequired,
+    handlePDFUpload: PropTypes.func.isRequired,
+    previewFile: PropTypes.object.isRequired,
+    create: PropTypes.func.isRequired,
+    done: PropTypes.func.isRequired
+  }
+
+  static defaultProps = {
+    previewFile: {}
+  }
+
+  renderFileUpload = (field, form) => {
+    const { name } = field
+    const { setFieldValue } = form
+    const {
+      handleImageUpload,
+      handlePDFUpload,
+      previewFile
+    } = this.props
+
+    return (
+      <Dropzone
+        name={name}
+        accept='application/pdf,image/jpeg'
+        style={{
+          alignItems: 'center',
+          cursor: 'pointer',
+          display: 'flex',
+          height: '250px',
+          justifyContent: 'center',
+          textAlign: 'center'
+        }}
+        activeStyle={{
+          borderColor: '#6c6',
+          backgroundColor: '#eee'
+        }}
+        rejectStyle={{
+          borderColor: '#c66',
+          backgroundColor: '#eee'
+        }}
+        className='form-control'
+        onDrop={(acceptedFiles) => {
+          const file = acceptedFiles[0] // Only 1 file per submission
+
+          switch (file.type) {
+            case 'application/pdf':
+              handlePDFUpload(file).then(() => {
+                // Need to use 'this.props' here to get the most up-to-date value – 'previewFile' above will be out-of-date
+                setFieldValue(name, this.props.previewFile.path)
+              })
+              break
+            case 'image/jpeg':
+              handleImageUpload(file).then(() => {
+                // Need to use 'this.props' here to get the most up-to-date value – 'previewFile' above will be out-of-date
+                setFieldValue(name, this.props.previewFile.path)
+              })
+              break
+            default:
```

Comment:
what are we doing about the video case?

Reply:
Students who want to submit a YouTube or Vimeo video will choose the "Video" Submission Form – this is the "Other Media" form where students can submit an image or PDF of something like a book or physical object.

We don't need to worry about videos on this form, as students cannot upload video files.